cogs/welcome/welcome.py

```diff
@@ -42,15 +52,72 @@ def __init__(self, bot: Red):
         self.config = Config.get_conf(self, identifier=5842647, force_registration=True)
         self.config.register_guild(**DEFAULT_GUILD)
 
+    async def getRandomMessage(self, guild):
+        greetings = await self.config.guild(guild).greetings()
+        numGreetings = len(greetings)
+        if not greetings:
+            return "Welcome to the server {USER}"
+        else:
+            return random.choice(list(greetings.values()))
+
     # The async function that is triggered on new member join.
     @commands.Cog.listener()
     async def on_member_join(self, newMember: discord.Member):
+        await self.sendWelcomeMessageChannel(newMember)
         await self.sendWelcomeMessage(newMember)
 
     @commands.Cog.listener()
     async def on_member_remove(self, leaveMember: discord.Member):
         await self.logServerLeave(leaveMember)
 
+    # This async function is to look for if the welcome channel was removed
+    @commands.Cog.listener()
+    async def on_channel_remove(self, removedChannel: discord.TextChannel):
+        guild = removedChannel.guild
+        # the channel to post welcome stuff in
+        welcomeIDSet = await self.config.guild(guild).welcomeChannelSet()
+        welcomeID = await self.config.guild(guild).welcomeChannel()
+        if welcomeIDSet and removedChannel.id == welcomeID:
+            await self.config.guild(guild).welcomeChannelSet.set(False)
+            LOGGER.info("Changed guild's welcomeChannelSetFlag to false as channel was deleted")
+        return
+
+    async def sendWelcomeMessageChannel(self, newUser: discord.Member):
+        guild = newUser.guild
+        channelID = await self.config.guild(guild).welcomeChannel()
+        isSet = await self.config.guild(guild).welcomeChannelSet()
+        # if channel isn't set
+        if not isSet:
+            return
+        channel = discord.utils.get(guild.channels, id=channelID)
+        rawMessage = await self.getRandomMessage(guild)
+
+        message = rawMessage.replace("{USER}", newUser.mention)
+        splitMessage = rawMessage.split("{USER}")
+
+        joinLogEnabled = await self.config.guild(newUser.guild).logJoinEnabled()
+        joinLogChannel = await self.config.guild(newUser.guild).logJoinChannel()
+        try:
+            await channel.send(message)
+        except (discord.Forbidden, discord.HTTPException) as errorMsg:
+            LOGGER.error(
+                "Could not send message, please make sure the bot "
+                "has enough permissions to send messages to this "
+                "channel!",
+                exc_info=True,
+            )
+            LOGGER.error(errorMsg)
+        else:
+            if joinLogEnabled:
+                LOGGER.info(
+                    "User %s#%s (%s) has joined. Posted welcome message.",
+                    newUser.name,
+                    newUser.discriminator,
+                    newUser.id,
+                )
+
```

Comment:
Let's log unconditionally: no need for joinLogEnabled.

Reply:
done